src/ets/scala/io/iohk/ethereum/ets/blockchain/scenario.scala

```diff
@@ -0,0 +1,66 @@
+package io.iohk.ethereum.ets.blockchain
+
+import akka.util.ByteString
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.ets.common.AccountState
+
+
+case class BlockchainScenario(
+   blocks: List[BlockDef],
+   genesisBlockHeader: BlockHeaderDef,
+   genesisRLP: Option[ByteString],
+   lastblockhash: ByteString,
+   network: String,
+   postState: Map[Address, AccountState],
+   pre: Map[Address, AccountState]
+)
+
+
+case class BlockDef(
+  rlp: String,
+  blocknumber: Option[BigInt],
+  blockHeader: Option[BlockHeaderDef],
+  transactions: Option[Seq[TransactionDef]],
+  uncleHeaders: Option[Seq[BlockHeaderDef]]
+)
+
+
+case class TransactionDef(
+  nonce: BigInt,
+  gasPrice: BigInt,
+  gasLimit: BigInt,
+  to: Option[Address],
+  value: BigInt,
+  data:ByteString,
+  r: BigInt,
+  s: BigInt,
+  v: ByteString
+)
+
+case class BlockHeaderDef(
```

Comment:
Minor comment: Could we use `BlockHeader` instead, as it is defined in the same way?

Reply:
We could, but I was rather insistent on not doing so. These classes implement a similar model, but in totally different contexts. The fact the implementation happen to be the same shouldn't drive the design. For reference/rationale see _bounded contexts_ principle in DDD.